Sources/NIO/SocketAddresses.swift

```diff
@@ -363,6 +363,35 @@ extension SocketAddress: Equatable {
     }
 }
 
+/// We define an extension on `SocketAddress` that gives it an elementwise hashable conformance, using
+/// only the elements defined on the structure in their man pages (excluding lengths).
+extension SocketAddress: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        switch self {
+        case .unixDomainSocket(let uds):
+            hasher.combine(0)
+            hasher.combine(uds.address.sun_family)
+            withUnsafeBytes(of: uds.address.sun_path) {
```

Comment:
This is actually incorrect in both hashable and `==`. This needs to only compare the bytes until the first `0` byte I think.

Reply:
(ignore this is we know for sure that all trailing bytes are always `0` for example  in which case we should add an `assert`).

Reply:
Yeah, I felt pretty weird about it as I was writing it tbh.

Reply:
Updated, please take a look.